tests/cases/conformance/classes/propertyMemberDeclarations/assignParameterPropertyToPropertyDeclarationESNext.ts

```diff
@@ -4,6 +4,7 @@ class C {
     qux = this.bar // should error
     bar = this.foo // should error
     quiz = this.bar // ok
+    quench = this.m1() // ok
```

Comment:
(Maybe add a similar case with `m3 = function() ...` which should error, as a reminder that it's different and in case some rewrites are added in the future?)